src/main/java/org/spongepowered/api/world/World.java

```diff
@@ -329,6 +329,25 @@ default void setSerializationBehavior(SerializationBehavior behavior) {
      */
     PortalAgent getPortalAgent();
 
+    /**
+     * Get the y value of the highest block that sunlight can reach in the given column.
+     * <p>This method ignores all transparent blocks, providing the highest opaque block.</p>
+     *
+     * @return The y value of the highest opaque block
+     */
+    int getHighestYAt(int x, int z);
+
+    /**
+     * Get the {@link Location} of the highest block that sunlight can reach in the given column.
+     * <p>This method ignores all transparent blocks, providing the highest opaque block.</p>
+     *
+     * @param position The column position
+     * @return The highest opaque block
+     */
+    default Location<World> getHighestBlockAt(Vector3i position) {
```

Comment:
"Well it makes it easier to do something like this world.getHighestBlockAt(location.getBlockPosition()). "